cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -651,7 +651,7 @@ func (scaleSet *ScaleSet) Nodes() ([]cloudprovider.Instance, error) {
 	curSize, err := scaleSet.getCurSize()
 	if err != nil {
 		klog.Errorf("Failed to get current size for vmss %q: %v", scaleSet.Name, err)
-		return nil, err
+		return nil, nil // Don't return error if VMSS not found
```

Comment:
This prevents us from failing to create AzureManager if one vmss is not found/not yet created.

Reply:
Could getCurSize return other kinds of errors, beside not found?

Reply:
That is actually a fair call. 

There is another error scenario where we fail to fetch latest info from VMSS client for spot instances:
https://github.com/kubernetes/autoscaler/blob/9b41b9dc5bf431a4ad657c044e809b15cc28a7e2/cluster-autoscaler/cloudprovider/azure/azure_scale_set.go#L221-L233

I need to think more on how to differentiate between the two. An ugly solution would be to return -2 for this one...lol

A better solution may be to have different error types to switch on:
1. not found via VMSS cache
2. not found via VMSS client

Reply:
I am not sure we need to treat these two cases differently, do we?